src/Microsoft.DotNet.Build.CloudTestTasks/DownloadFromAzure.cs

```diff
@@ -95,10 +96,16 @@ public override bool Execute()
                         {
                             blobDirectory = blob.Substring(0, dirIndex);
                             blobFilename = blob.Substring(dirIndex + 1);
-                        }
-                        if(BlobNamePrefix != null)
-                        {
-                            blobDirectory = blobDirectory.Substring(BlobNamePrefix.Length);
+
+                            // Trim blob name prefix (directory part) from download to blob directory
+                            if(BlobNamePrefix != null)
+                            {
+                                if(BlobNamePrefix.Length > dirIndex)
+                                {
+                                    BlobNamePrefix = BlobNamePrefix.Substring(0, dirIndex);
```

Comment:
This seems like a hacky workaround;  if you're going to do this I'd prefer we also at least:

1) Provide a warning if the blob name doesn't start with the supplied prefix
2) Totally don't use the prefix at all in this case.

I get what happened is that we specified the prefix as Release/pkg/Microsoft but it only used Release/pkg, but this is a user-error case;  if the user makes another error and has total garbage in this property,  you just take dirIndex random characters off that string, at which point you could just not use the prefix at all and just the index value.